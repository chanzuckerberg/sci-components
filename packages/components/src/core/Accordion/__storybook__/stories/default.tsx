import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
