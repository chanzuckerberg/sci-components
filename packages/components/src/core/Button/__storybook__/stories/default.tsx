import { Args } from "@storybook/react";
import RawButton from "src/core/Button";

export const Button = (props: Args): JSX.Element => {
  const { sdsType, sdsStyle, text, startIcon, startIċon, endIcon, endIċon } =
    props;

  const startIconFinal = startIcon || startIċon;
  const endIconFinal = endIcon || endIċon;

  return (
    <RawButton
      sdsType={sdsType}
      sdsStyle={sdsStyle}
      startIcon={startIconFinal}
      endIcon={endIconFinal}
      {...props}
    >
      {text}
    </RawButton>
  );
};
