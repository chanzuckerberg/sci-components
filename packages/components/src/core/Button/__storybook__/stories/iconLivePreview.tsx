import { SDSAppTheme } from "src/core/styles";
import RawButton from "src/core/Button";
// import { BUTTON_TEXT, DEFAULT_PLACEMENT_STYLES } from "../constants";
// import Icon from "src/core/Icon";

// Icon Live Preview Demo

export const IconLivePreviewDemo = (): JSX.Element => {
  const spaces = SDSAppTheme?.spacing;

  const livePreviewStyles: React.CSSProperties = {
    alignItems: "center",
    display: "grid",
    gridColumnGap: "24px",
    gridTemplateColumns: "repeat(7, min-content)",
  };

  return (
    <div style={livePreviewStyles}>
      <div style={{ display: "flex" }}>
        <RawButton
          aria-label="grid"
          style={{ marginRight: spaces?.xxs }}
          icon="Grid"
          sdsSize="large"
          sdsType="primary"
          sdsStyle="icon"
        />
        <RawButton
          aria-label="grid"
          style={{ marginRight: spaces?.xxs }}
          icon="Grid"
          sdsSize="large"
          sdsType="primary"
          sdsStyle="icon"
        />
      </div>
      <div style={{ display: "flex" }}>
        <RawButton
          aria-label="infoSpeechBubble"
          style={{ marginRight: spaces?.m }}
          icon="InfoSpeechBubble"
          sdsSize="large"
          sdsType="secondary"
          sdsStyle="icon"
        />
        <RawButton
          aria-label="infoSpeechBubble"
          style={{ marginRight: spaces?.m }}
          icon="InfoSpeechBubble"
          sdsSize="large"
          sdsType="secondary"
          sdsStyle="icon"
        />
      </div>
      <div>
        <RawButton
          aria-label="xMark"
          style={{ marginRight: spaces?.m }}
          icon="XMark"
          sdsSize="large"
          sdsType="tertiary"
          sdsStyle="icon"
        />
      </div>
      <div>
        <RawButton
          aria-label="xMark"
          style={{ marginRight: spaces?.m }}
          icon="XMark"
          sdsSize="medium"
          sdsType="tertiary"
          sdsStyle="icon"
        />
      </div>
      <div style={{ display: "flex" }}>
        <RawButton
          aria-label="barChartVertical3"
          style={{ marginRight: spaces?.s }}
          icon="BarChartVertical3"
          sdsSize="small"
          sdsType="primary"
          sdsStyle="icon"
        />
        <RawButton
          aria-label="barChartVertical3"
          style={{ marginRight: spaces?.s }}
          icon="BarChartVertical3"
          sdsSize="small"
          sdsType="primary"
          sdsStyle="icon"
        />
      </div>
      <div style={{ display: "flex" }}>
        <RawButton
          aria-label="plusCircle"
          style={{ marginRight: spaces?.s }}
          icon="PlusCircle"
          sdsSize="small"
          sdsType="secondary"
          sdsStyle="icon"
        />
        <RawButton
          aria-label="plusCircle"
          style={{ marginRight: spaces?.s }}
          icon="PlusCircle"
          sdsSize="small"
          sdsType="secondary"
          sdsStyle="icon"
        />
      </div>
      <div>
        <RawButton
          aria-label="xMark"
          style={{ marginRight: spaces?.s }}
          icon="XMark"
          sdsSize="small"
          sdsType="tertiary"
          sdsStyle="icon"
        />
      </div>
    </div>
  );
};
