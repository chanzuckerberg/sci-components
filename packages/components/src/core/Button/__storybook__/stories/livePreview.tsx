import { Args } from "@storybook/react";
import RawButton from "src/core/Button";
import {
  BUTTON_TEXT,
  DEFAULT_PLACEMENT_STYLES,
  MINIMAL_PLACEMENT_STYLES,
} from "../constants";
import Icon from "src/core/Icon";

// Rounded Live Preview Demo

export const RoundedLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={DEFAULT_PLACEMENT_STYLES}>
      <div>
        <RawButton {...props} sdsStyle="rounded" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="xl" sdsType="button" />}
          sdsStyle="rounded"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton {...props} sdsStyle="rounded" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="s" sdsType="button" />}
          sdsStyle="rounded"
          sdsType="secondary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};

// Square Live Preview Demo

export const SquareLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={DEFAULT_PLACEMENT_STYLES as React.CSSProperties}>
      <div>
        <RawButton {...props} sdsStyle="square" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="l" sdsType="button" />}
          sdsStyle="square"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton {...props} sdsStyle="square" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="l" sdsType="button" />}
          sdsStyle="square"
          sdsType="secondary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};

// Minimal Live Preview Demo

export const MinimalLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={MINIMAL_PLACEMENT_STYLES}>
      <div>
        <RawButton {...props} sdsStyle="minimal" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="s" sdsType="button" />}
          sdsStyle="minimal"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton {...props} sdsStyle="minimal" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};
