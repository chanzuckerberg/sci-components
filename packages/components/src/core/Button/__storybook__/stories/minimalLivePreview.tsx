import { Args } from "@storybook/react";
import RawButton from "src/core/Button";
import { BUTTON_TEXT, MINIMAL_PLACEMENT_STYLES } from "../constants";
import Icon from "src/core/Icon";

// Minimal Live Preview Demo

export const MinimalLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={MINIMAL_PLACEMENT_STYLES}>
      <div>
        <RawButton {...props} sdsStyle="minimal" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="s" sdsType="button" />}
          sdsStyle="minimal"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton {...props} sdsStyle="minimal" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};
