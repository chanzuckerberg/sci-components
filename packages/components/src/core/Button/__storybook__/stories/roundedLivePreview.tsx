import { Args } from "@storybook/react";
import RawButton from "src/core/Button";
import { BUTTON_TEXT, DEFAULT_PLACEMENT_STYLES } from "../constants";
import Icon from "src/core/Icon";

// Rounded Live Preview Demo

export const RoundedLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={DEFAULT_PLACEMENT_STYLES}>
      <div>
        <RawButton {...props} sdsStyle="rounded" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="xl" sdsType="button" />}
          sdsStyle="rounded"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton {...props} sdsStyle="rounded" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>
      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="s" sdsType="button" />}
          sdsStyle="rounded"
          sdsType="secondary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};
