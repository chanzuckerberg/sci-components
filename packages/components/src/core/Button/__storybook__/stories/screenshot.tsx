import { Args } from "@storybook/react";
import RawButton from "src/core/Button";
import {
  BUTTON_DISABLED_OPTIONS,
  BUTTON_ICON_OPTIONS,
  BUTTON_PSEUDO_STATES,
  BUTTON_SDS_STYLES,
  BUTTON_SDS_TYPES,
  BUTTON_TEXT,
} from "../constants";

export const ScreenshotTestDemo = (props: Args): JSX.Element => {
  const topLevel: React.CSSProperties = {
    columnGap: "20px",
    display: "inline-grid",
    fontFamily: "sans-serif",
    marginRight: "50px",
  };
  const displayContents: React.CSSProperties = {
    display: "contents",
  };
  const penultimateLevel: React.CSSProperties = {
    display: "contents",
  };
  const bottomLevel: React.CSSProperties = {
    marginBottom: 10,
  };
  const fontWeightNormal: React.CSSProperties = {
    fontWeight: "normal",
  };
  const topLabel: React.CSSProperties = {
    ...fontWeightNormal,
    fontSize: "2em",
    gridColumn: "1 / 6",
    marginBottom: 0,
  };
  const midLabel: React.CSSProperties = {
    ...fontWeightNormal,
    borderStyle: "solid none none none",
    gridColumn: "1 / 6",
    justifySelf: "stretch",
    paddingTop: 10,
  };
  const secondLabel: React.CSSProperties = {
    ...midLabel,
    borderWidth: "2px",
    fontSize: "1.17em",
    margin: "20px 0",
  };
  const thirdLabel: React.CSSProperties = {
    ...midLabel,
    alignSelf: "end",
    borderWidth: "1px",
    fontWeight: "normal",
    margin: "0 0 5px 0",
  };
  const bottomLabel: React.CSSProperties = {
    ...fontWeightNormal,
    margin: "10px 0",
  };

  // loop through all BUTTON_SDS_STYLES
  return (
    <>
      {BUTTON_SDS_STYLES.map((sdsStyle) => {
        return <ButtonStyleOption sdsStyle={sdsStyle} key={sdsStyle} />;
      })}
    </>
  );

  // loop through all BUTTON_SDS_TYPES + create headers for BUTTON_SDS_STYLES
  function ButtonStyleOption({
    sdsStyle,
  }: {
    sdsStyle: (typeof BUTTON_SDS_STYLES)[number];
  }) {
    return (
      <div style={topLevel}>
        <h3 style={topLabel}>
          Style: <b>{sdsStyle}</b>
        </h3>
        {BUTTON_SDS_TYPES.map((type) => {
          return (
            <ButtonTypeOption sdsStyle={sdsStyle} type={type} key={type} />
          );
        })}
      </div>
    );
  }

  // loop through all BUTTON_ICON_OPTIONS + create headers for BUTTON_SDS_TYPES
  function ButtonTypeOption({
    sdsStyle,
    type,
  }: {
    sdsStyle: (typeof BUTTON_SDS_STYLES)[number];
    type: (typeof BUTTON_SDS_TYPES)[number];
  }) {
    return (
      <div style={displayContents}>
        <h4 style={secondLabel}>
          Type: <b>{type}</b>
        </h4>
        {/* Minimal Secondary doesn't have icon button option */}
        {sdsStyle === "minimal" && type === "secondary" ? (
          <ButtonIconOption
            sdsStyle={sdsStyle}
            type={type}
            icon={BUTTON_ICON_OPTIONS[0]}
            key={String(BUTTON_ICON_OPTIONS[0])}
          />
        ) : (
          BUTTON_ICON_OPTIONS.map((icon) => {
            return (
              <ButtonIconOption
                sdsStyle={sdsStyle}
                type={type}
                icon={icon}
                key={String(icon)}
              />
            );
          })
        )}
      </div>
    );
  }

  // loop through all BUTTON_DISABLED_OPTIONS + create headers for BUTTON_ICON_OPTIONS
  function ButtonIconOption({
    sdsStyle,
    type,
    icon,
  }: {
    sdsStyle: (typeof BUTTON_SDS_STYLES)[number];
    type: (typeof BUTTON_SDS_TYPES)[number];
    icon: (typeof BUTTON_ICON_OPTIONS)[number];
  }) {
    return (
      <div style={displayContents}>
        <h5 style={thirdLabel}>
          Icon: <b>{icon ? "yes" : "no"}</b>
        </h5>
        {BUTTON_DISABLED_OPTIONS.map((disabled) => {
          return (
            <ButtonDisabledOption
              sdsStyle={sdsStyle}
              type={type}
              icon={icon}
              disabled={disabled}
              key={String(disabled)}
            />
          );
        })}
      </div>
    );
  }

  // loop through all BUTTON_PSEUDO_STATES + create headers for BUTTON_DISABLED_OPTIONS, BUTTON_PSEUDO_STATES
  function ButtonDisabledOption({
    sdsStyle,
    type,
    icon,
    disabled,
  }: {
    sdsStyle: (typeof BUTTON_SDS_STYLES)[number];
    type: (typeof BUTTON_SDS_TYPES)[number];
    icon: (typeof BUTTON_ICON_OPTIONS)[number];
    disabled: (typeof BUTTON_DISABLED_OPTIONS)[number];
  }) {
    return (
      <div style={penultimateLevel}>
        {BUTTON_PSEUDO_STATES.map((state) => {
          return (
            <div style={bottomLevel} key={state}>
              {/* removes irrelevant disabled iterations: when combined with all pseudo-states except default, `disabled=false` is impossible */}
              {(disabled === false ||
                (disabled === true && state === "default")) && (
                <>
                  <h6 style={bottomLabel}>
                    {disabled === false ? "State: " : "Disabled: "}
                    <br />
                    <b>{disabled === false ? state : "true"}</b>
                  </h6>
                  <RawButton
                    {...props}
                    data-testid="button"
                    sdsStyle={sdsStyle}
                    sdsType={type}
                    startIcon={icon}
                    disabled={disabled}
                    className={`pseudo-${state}`}
                    key={state}
                  >
                    {BUTTON_TEXT}
                  </RawButton>
                </>
              )}
            </div>
          );
        })}
      </div>
    );
  }
};
