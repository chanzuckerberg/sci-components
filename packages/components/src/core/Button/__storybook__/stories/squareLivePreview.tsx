import { Args } from "@storybook/react";
import RawButton from "src/core/Button";
import { BUTTON_TEXT, DEFAULT_PLACEMENT_STYLES } from "../constants";
import Icon from "src/core/Icon";

// Square Live Preview Demo

export const SquareLivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={DEFAULT_PLACEMENT_STYLES as React.CSSProperties}>
      <div>
        <RawButton {...props} sdsStyle="square" sdsType="primary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="l" sdsType="button" />}
          sdsStyle="square"
          sdsType="primary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton {...props} sdsStyle="square" sdsType="secondary">
          {BUTTON_TEXT}
        </RawButton>
      </div>

      <div>
        <RawButton
          {...props}
          startIcon={<Icon sdsIcon="Download" sdsSize="l" sdsType="button" />}
          sdsStyle="square"
          sdsType="secondary"
        >
          {BUTTON_TEXT}
        </RawButton>
      </div>
    </div>
  );
};
