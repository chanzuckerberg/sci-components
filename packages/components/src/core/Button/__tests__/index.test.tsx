import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "../__storybook__/index.stories";

const { Test } = composeStories(stories);

describe("<button />", () => {
  generateSnapshots(stories);

  it("renders Button component", () => {
    render(<Test />);
    const elements = screen.getAllByTestId("button");
    expect(elements.length).toBeTruthy();
  });
});
