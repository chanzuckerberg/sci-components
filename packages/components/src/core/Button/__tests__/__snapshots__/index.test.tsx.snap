// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<button /> Default story renders snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-19167ln-MuiButtonBase-root-MuiButton-root"
  icon="Cube"
  tabindex="0"
  text="Label"
  type="button"
>
  Label
</button>
`;
